File: sandbox/fmc/fmc-rest/src/main/scala/org/fusesource/fabric/webui/agents/osgi/ServiceResource.scala
/*
 * Copyright 2010 Red Hat, Inc.
 *
 *  Red Hat licenses this file to you under the Apache License, version
 *  2.0 (the "License"); you may not use this file except in compliance
 *  with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
 *  implied.  See the License for the specific language governing
 *  permissions and limitations under the License.
 */
package io.fabric8.webui.agents.osgi

import io.fabric8.api.data.ServiceInfo
import org.codehaus.jackson.annotate.JsonProperty
import io.fabric8.api.data.ServiceInfo.Property
import io.fabric8.webui.BaseResource

class PropertyResource(property: Property) {
  @JsonProperty
  def key = property.getKey

  @JsonProperty
  def value = property.getValue
}

class ServiceResource(val service: ServiceInfo) extends BaseResource {

  @JsonProperty
  def id = service.getId.longValue

  @JsonProperty
  def bundle_id = service.getBundleId.longValue

  @JsonProperty
  def object_classes = service.getObjectClasses

  @JsonProperty
  def using_bundles = service.getUsingBundlesId map (_.longValue)

  @JsonProperty
  def properties = service.getProperties.map(new PropertyResource(_))

}

File: components/gateway-apiman/src/main/java/io/fabric8/gateway/apiman/rest/ServiceResource.java
/*
 *  Copyright 2005-2015 Red Hat, Inc.
 *
 *  Red Hat licenses this file to you under the Apache License, version
 *  2.0 (the "License"); you may not use this file except in compliance
 *  with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
 *  implied.  See the License for the specific language governing
 *  permissions and limitations under the License.
 */
package io.fabric8.gateway.apiman.rest;

import io.apiman.gateway.api.rest.contract.IServiceResource;
import io.apiman.gateway.api.rest.contract.exceptions.NotAuthorizedException;
import io.apiman.gateway.engine.beans.Service;
import io.apiman.gateway.engine.beans.ServiceEndpoint;
import io.apiman.gateway.engine.beans.exceptions.PublishingException;
import io.fabric8.gateway.apiman.ApiManEngine;
/**
 * Implements the IServiceResource interface to set up a REST
 * interface to configure Services in APIMan Engine.
 */
public class ServiceResource implements IServiceResource {

	private ApiManEngine engine;
	
	public ServiceResource(ApiManEngine engine) {
		super();
		this.engine = engine;
	}

	@Override
	public void publish(Service service) throws PublishingException,
			NotAuthorizedException {
		engine.publishService(service);
	}

	@Override
	public void retire(String organizationId, String serviceId, String version) throws PublishingException,
			NotAuthorizedException {
		engine.retireService(organizationId, serviceId, version);
	}

	@Override
	public ServiceEndpoint getServiceEndpoint(String orgId, String serviceId, String version) 
			throws NotAuthorizedException {
		ServiceEndpoint serviceEndpoint = new ServiceEndpoint();
		serviceEndpoint.setEndpoint(engine.serviceMapping(orgId, serviceId, version));
		return serviceEndpoint;
	}

}
